src/sentry/preprod/api/endpoints/project_preprod_artifact_install_details.py

```diff
@@ -0,0 +1,138 @@
+
+from sentry import analytics
+from sentry.api.api_owners import ApiOwner
+from sentry.api.api_publish_status import ApiPublishStatus
+from sentry.api.base import region_silo_endpoint
+from sentry.api.bases.project import ProjectEndpoint
+from sentry.preprod.models import InstallablePreprodArtifact, PreprodArtifact
+
+logger = logging.getLogger(__name__)
+
+
+def create_installable_preprod_artifact(
+    preprod_artifact: PreprodArtifact, expiration_hours: int = 12
+) -> InstallablePreprodArtifact:
+    """
+    Creates a new InstallablePreprodArtifact for a given PreprodArtifact.
+
+    Args:
+        preprod_artifact: The PreprodArtifact to create an installable version for
+        expiration_hours: Number of hours until the install link expires (default: 24)
+
+    Returns:
+        The created InstallablePreprodArtifact instance
+    """
+
+    url_path = secrets.token_urlsafe(12)
+
+    # Set expiration date
+    expiration_date = timezone.now() + timedelta(hours=expiration_hours)
+
+    # Create the installable artifact
+    installable_artifact = InstallablePreprodArtifact.objects.create(
+        preprod_artifact=preprod_artifact,
+        url_path=url_path,
+        expiration_date=expiration_date,
+        download_count=0,
+    )
+
+    logger.info(
+        "Created installable preprod artifact",
+        extra={
+            "installable_artifact_id": installable_artifact.id,
+            "preprod_artifact_id": preprod_artifact.id,
+            "project_id": preprod_artifact.project.id,
+            "organization_id": preprod_artifact.project.organization.id,
+            "url_path": url_path,
+            "expiration_date": expiration_date.isoformat(),
+        },
+    )
+
+    return installable_artifact
+
+
+@region_silo_endpoint
+class ProjectPreprodInstallDetailsEndpoint(ProjectEndpoint):
+    owner = ApiOwner.EMERGE_TOOLS
+    publish_status = {
+        "GET": ApiPublishStatus.EXPERIMENTAL,
+    }
+
+    def get(self, request: Request, project, artifact_id) -> HttpResponseBase:
+        analytics.record(
+            "preprod_artifact.api.install_details",
+            organization_id=project.organization_id,
+            project_id=project.id,
+            user_id=request.user.id,
+            artifact_id=artifact_id,
+        )
+
+        try:
+            preprod_artifact = PreprodArtifact.objects.get(
+                project=project,
+                id=artifact_id,
+            )
+        except PreprodArtifact.DoesNotExist:
+            return Response({"error": "Preprod artifact not found"}, status=404)
+
+        # For iOS apps (XCARCHIVE), check code signature validity
+        if preprod_artifact.artifact_type == PreprodArtifact.ArtifactType.XCARCHIVE:
+            if (
+                not preprod_artifact.extras
+                or preprod_artifact.extras.get("is_code_signature_valid") is not True
+            ):
+                return JsonResponse(
+                    {
```

Comment:
Just thinking ahead, is Android going to have a field like `is_universal_apk`, and the frontend will check the combination of booleans?